contracts/contracts/SNStateProofVerifier.sol

```diff
@@ -248,15 +269,48 @@ contract SNStateProofVerifier is
         return aExtracted == b;
     }
 
+    // overloaded/wrapper function around verifyProof, used to verify storage proof array where the class commitment does not apply
+    function verifyStorageProof(
+        uint256 rootHash,
+        uint256 path,
+        StarknetProof[] calldata proofArray
+    ) public view returns (uint256 value) {
+        // classCommitment is 0 means we are verifying storage proof array
+        return verifyProof(rootHash, path, proofArray, 0);
+    }
+
     // A generic method to verify a proof against a root hash and a path.
     function verifyProof(
         uint256 rootHash,
         uint256 path,
-        StarknetProof[] calldata proofArray
+        StarknetProof[] calldata proofArray,
+        uint256 classCommitment // 0 means no class commitment
     ) public view returns (uint256 value) {
         uint256 expectedHash = rootHash;
         int256 pathBitIndex = 250; // start from the MSB bit index
+        if (classCommitment > 0) {
+            // https://docs.starknet.io/documentation/architecture_and_concepts/State/starknet-state/
+            require(
+                proofArray.length > 0,
+                "proofs must have atleast one element!"
+            );
 
+            uint256 calculatedContractStateRoot = hashForSingleProofNode(
+                proofArray[0]
+            ); // the hash of first element in the proof array is the contract state root
+            uint256[] memory poseidonInput = new uint256[](3);
+            poseidonInput[0] = FELT_FOR_STARKNET_STATE_V0;
+            poseidonInput[1] = calculatedContractStateRoot;
+            poseidonInput[2] = classCommitment;
+
+            uint256 calculateStateCommitment = poseidonHashMany(poseidonInput);
+            require(
+                calculateStateCommitment == expectedHash,
+                "calculated state commitment doesn't match with the expected state commitment!"
+            );
+            // since we have already verified the first element in the proof array correctly hashes up to the state commitment, we can assume the hash of first element in the proof array is correct.
+            expectedHash = calculatedContractStateRoot;
+        }
         require(
```

Comment:
we can move this check before the if statement

Reply:
done